gdal/swig/java/GNUmakefile

```diff
@@ -99,8 +100,8 @@ test:
 	${JAVA_RUN} ogrtindex tmp_test/contour_index.shp tmp_test/contour.shp
 	${JAVA_RUN} OSRTest
 
-$(JAVA_MODULES): $(JAVA_OBJECTS)
-	$(LINK) $(LDFLAGS) $^ $(CONFIG_LIBS) -o $@ $(LINK_EXTRAFLAGS)
+$(JAVA_MODULES): $(JAVA_OBJECTS) 
+	$(LINK) $(LDFLAGS) $^ $(GDAL_LIB) $(CONFIG_LIBS) -o $@ $(LINK_EXTRAFLAGS)
```

Comment:
I'm not sure this is a good idea to incorporate libgdal in the java bindings.

Reply:
Well, do you think it will cause a problem?  Personally, I prefer to have only one file to deal with.  But, certainly, back out the change if you have strong feelings about it.

Note that if the Windows version is built with DLLBUILD undefined, this is what happens with the unmodified code base, with all of the referenced objects being included in each of the DLL's that are built.

Reply:
Well perhaps this makes sense on Windows, but this is less in the Unix philosophy to bundle everything

Reply:
I'll let you decide.
My interest is upgrading the GDAL library included with Nasa's WorldWind.  The copy of GDAL they are including is version 1.7.2.  They have released copies of the JNI interface for multiple platforms with libgdal included in the JNI shared library.  It seemed like a good idea to me, but for the life of me I can't see how they included the MrSID libraries in their DLL or .so shared libraries.
They also installed a custom loader.  I have a parallel pull request outstanding with them (https://github.com/NASAWorldWind/WorldWindJava/pull/146), but I left the libraries separate, thinking that if they were to be combined that should be part of GDAL.

Happy Holidays!

Reply:
> I'll let you decide.

I'll prefer to not have libgdal incorporated in the libgdaljniall.so, at least not by default. If you can find a way of parameterizing it, why not.

Reply:
Does that last change meet with your approval?